livekit-agents/livekit/agents/utils/codecs/decoder.py

```diff
@@ -92,13 +93,16 @@ class AudioStreamDecoder:
     _max_workers: int = 10
     _executor: Optional[ThreadPoolExecutor] = None
 
-    def __init__(self, *, sample_rate: int = 48000, num_channels: int = 1):
+    def __init__(
+        self, *, sample_rate: int = 48000, num_channels: int = 1, format: str | None = None
```

Comment:
Can we remove the format? and auto-detect?

Reply:
i don't believe we can very easily, it'll make it pretty messy to feed it to ffmpeg containers after already consuming part of the input

Reply:
It's unclear when you need to provide a format vs not?

Reply:
it's a good practice to provide the format because it's known. and plugins definitely know the format because they have to ask the API to return content as such

Reply:
can we use the format for all plugins? the format isn't passed into av.open

Reply:
it doesn't add immediate value.. that's why it's optional. this only matters when it's wav.. that's the point of this PR.